docs_src/design/ucr/0001-System-Events-for-Devices.md

```diff
@@ -0,0 +1,49 @@
+
+### Target Users
+- Device User
+- Software Deplorer
+- Software Integrator
+- Deployment Customer
+
+### Description
+System Events (aka Control Plane Events - CPE) are events generated by the system when there are changes in part of the system that are important for other parts of the system to know about. This UCR will focus on the Device System Events use case as related to camera management. These Device System Events could be utilized by many other use cases in similar manner.
+
+The new EdgeX USB and ONVIF camera Device Services (not yet released) implement auto provisioning which detects when a new camera device has been connected to USB or added to the network. New Device objects are created in Core Metadata for the new camera devices that have been auto provisioned. The auto provisioning also detects existing known camera devices, determines if there have been changes to the device details, such as IP address, and updates the Device object in Core Metadata with any changes. Device objects can also be manually deleted from Core Metadata once camera devices have been permanently disconnected.
+
+A camera management application service needs to know when a new camera has been added so that it can initiate AI/ML processing on the stream from the new camera. The service also needs to know when an existing camera device has been updated so that it can make any needed adjustments such as restart the AI/ML processing using the new IP address of the camera. Finally the service needs to know when an existing camera device has been removed so that it can stop the AI/ML processing for the removed camera.
+
+### Existing solutions
+Parts of the system (i.e. application service) must poll Core Metadata for list of devices to determine if a device has been added, update or deleted. To do this it must keep its own list or Device objects for the know devices to make these determinations.
+
+As a temporary stop gap for the initial upcoming release of the new ONVIF camera device service, an enhancement was added which publishes an EdgeX Event/Reading when a new camera device has been added, updated or modified. The Reading contains the information about the event type and the device name. This is improper use of the EdgeX Event/Reading which is intended for readings from devices, not System Events. This feature in the ONVIF Camera Device Service will be removed once System Events for Devices are in place.
+
+### Requirements
+- Added/Updated/Deleted Device System Events are created and sent in a manner that other parts of the system can subscribe and receive them.
+
+- Each System Event must contain at a minimum the following:
+
+  - Event Type: Added, Updated or Deleted
+  - Device Name
+    - All that is needed to send a command to the device to get the stream URL or stop the AI/ML processing
```

Comment:
I think these could easy turn to feature requests issues (as [proposed in the template](https://github.com/edgexfoundry/edgex-docs/pull/775)) on the corresponding projects to have awareness of the existing workarounds (suggested here) and keep track of when the proposed system events get implemented.

Reply:
This assume where/how this gets implemented. i.e. messagebus vs support notifications. IMO the issues should not be created until the ADR is approved. At that point you know where the implementation will exist.

Reply:
> Feature requests don't specify how something should be implemented internally.

They do assume which repo the feature is needed. Which is some UCR is appropriate like when the request has come from community (not TSC). That is why I didn't push back to having the link in the template. IMO, the templates are guidelines and not every UCR/ADR will need every part of the template. I'll comment on the templet that the links should be stated as optional.

IMO (still), the issues for enhancements like this should be created once the ADR has been approved.

Reply:
> They do assume which repo the feature is needed. Which is some UCR is appropriate like when the request has come from community (not TSC). That is why I didn't push back to having the link in the template. IMO, the templates are guidelines and not every UCR/ADR will need every part of the template. I'll comment on the templet that the links should be stated as optional.
> 

I assumed that the feature requests submitted by the wider community are usually submitted to the application repos. Someone with deeper technical knowledge may be able to submit exactly where it will be implemented.

> IMO (still), the issues for enhancements like this should be created once the ADR has been approved.

In my mind, there are feature requests that may preexist or not even need an ADR. The submitter shouldn't have to know about architecture and design. I think you are referring to issues created to track a planned implementation.

Anyway, I've updated the template to permit this kind of use.